src/fsharp/SyntaxTree.fsi

```diff
@@ -24,22 +28,31 @@ type LongIdent = Ident list
 /// if dotRanges.Length = lid.Length, then the parser must have reported an error, so the typechecker is allowed
 /// more freedom about typechecking these expressions.
 /// LongIdent can be empty list - it is used to denote that name of some AST element is absent (i.e. empty type name in inherit)
-type LongIdentWithDots =
-    | //[<Experimental("This construct is subject to change in future versions of FSharp.Compiler.Service and should only be used if no adequate alternative is available.")>]
-      LongIdentWithDots of id: LongIdent * dotRanges: range list
+type LongIdentWithTrivia =
+    | LongIdentWithTrivia of id: LongIdent * dotRanges: range list * trivia: IdentTrivia option list
 
     /// Gets the syntax range of this construct
     member Range: range
 
     /// Get the long ident for this construct
-    member Lid: LongIdent
+    member LongIdent: LongIdent
+    
+    /// Get the idents with potential trivia attached
+    member IdentsWithTrivia: IdentWithTrivia list
 
     /// Indicates if the construct ends in '.' due to error recovery
     member ThereIsAnExtraDotAtTheEnd: bool
 
     /// Gets the syntax range for part of this construct
     member RangeWithoutAnyExtraDot: range
 
+[<AutoOpen>]
+module LongIdentWithTriviaHelpers =
+    val (|LongIdentWithDots|): LongIdentWithTrivia -> LongIdent * range list
```

Comment:
I think mark this as obsolete too, asking people either to define the active pattern in their own code or else use the members on the type.